### ISSUE TITLE
`reportUnnecessaryComparison` doesn't trigger when comparing string literal variable to the same string literal

### ISSUE THREAD

**Issue by injust** [DetachHead/basedpyright#1654]
### Description

https://basedpyright.com/?typeCheckingMode=all&code=IYAgvCBEBmD2uQFCIA4CcCWA7ALgClDAhnkgEpVNcCQBCYuBC9bfQ4gI2DXMtZvpQuPCkA

```python
a = "foo"

print(a == "foo")
print(a != "foo")
print(a == "bar")
print(a != "bar")
```

`reportUnnecessaryComparison` doesn't trigger when comparing `a` to `"foo"`.
